src/tactic/lift.lean

```diff
@@ -70,6 +70,9 @@ if h_some : h.is_some then
   -/
 meta def lift (p : pexpr) (t : pexpr) (h : option pexpr) (n : list name) : tactic unit :=
 do
+  is_prop_target ← target >>= is_prop,
```

Comment:
These three lines can become
`propositional_goal <|> fail "lift tactic failed. Tactic is only applicable when the target is a proposition.",`

Reply:
Ah nice. I tried to search for this, but couldn't find it because the tactic doesn't use `target` but `get_goals`.

Reply:
IIRC there's a reason why `is_prop_target isn't `target >>= is_prop >>= guardb`, but I can't remember why.... Probably should have been added as a comment.

Reply:
@semorrison is there a reason why `propositional_goal` uses `get_goals` and not `target` (related PRs: #125 and #1552)?